cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -229,6 +229,8 @@ public TwillController launch(TwillApplication twillApplication, Iterable<String
             // TokenSecureStoreUpdater.update() ignores parameters
             twillPreparer.addSecureStore(secureStoreUpdater.update(null, null));
           }
+
+          System.setProperty("user.name", cConf.get(Constants.CFG_HDFS_USER));
```

Comment:
@chtyim Any reason for not doing the same for release/3.3 branch as well - copy the `FileContextLocation` from twill and change the implementation of `getHomeLocation`